projects/cm_mcu/LocalTasks.c

```diff
@@ -335,20 +338,77 @@ struct MonitorI2CTaskArgs_t clockr0a_args = {
     .stack_size = 4096U,
 };
 
+void setFFmask(uint32_t present_FFLDAQ_F1, uint32_t present_FFL12_F1, uint32_t present_FFLDAQ_F2, uint32_t present_FFL12_F2)
+{
+  uint32_t whichff = 0;
+  uint64_t addr_ff = 0x44; // internal eeprom block for ff mask
+  uint32_t data = 0;
+  uint64_t pass = 0x12345678;
+
+  log_info(LOG_SERVICE, "Setting a bit mask of enabled Fireflys to 1 \r\n");
+
+  for (; whichff < NFIREFLIES; ++whichff) {
+    uint8_t val;
+    uint32_t shift;
+    if (whichff < NFIREFLIES_IT_F1) {
+      shift = whichff;
+      val = (present_FFL12_F1 >> (shift)) & 0x01;
+    }
+
+    else if (NFIREFLIES_IT_F1 <= whichff && whichff < NFIREFLIES_IT_F1 + NFIREFLIES_DAQ_F1) {
+      shift = whichff - NFIREFLIES_IT_F1 + 4;
+      val = (present_FFLDAQ_F1 >> (shift)) & 0x01;
+    }
+
+    else if (NFIREFLIES_F1 <= whichff && whichff < NFIREFLIES_F1 + NFIREFLIES_IT_F2) {
+      shift = whichff - NFIREFLIES_F1;
+      val = (present_FFL12_F2 >> (shift)) & 0x01;
+    }
+
+    else if (NFIREFLIES_F1 + NFIREFLIES_IT_F2 <= whichff) {
+      shift = whichff - NFIREFLIES_F1 - NFIREFLIES_IT_F2 + 4;
+      val = (present_FFLDAQ_F2 >> (shift)) & 0x01;
+    }
+
+    else {
+      val = 0;
+    }
+
+    log_debug(LOG_SERVICE, "%17s: %x", ff_moni2c_addrs[whichff].name, val);
+    data += (val << whichff);
+  }
+
+  data = (~data) & 0xffff;
+  ff_PRESENT_mask = data;
+  ff_USER_mask = data;
```

Comment:
I'm confused -- are you overwriting the value stored in the EEPROM with the same value as the PRESENT mask? I thought we were supposed to be able to manipulate these independently.

Reply:
I would have thought that ff_USER_mask is the value stored in the internal EEPROM.

Reply:
the reason that I set `ff_USER_mask` the same as `ff_PRSENT_mask` here is because this is at least a valid initial value for the `ff_USER_mask` variable before a user would register the board via `first_mcu` and change it to be whatever is stored in the internal EEPROM (https://github.com/apollo-lhc/cm_mcu/pull/145/files#diff-9cb461e4d0d68a2faa7e67efb69c308c2d463228064bb42ac14abd5e01af4a5fR862)

Reply:
but you never read it from the EEPROM do you?

Reply:
`ff_USER_mask` is only read from the EEPROM whenever a user writes a mask via e.g. `first_mcu 2 1 0 0`, where the first two are board num/rev and the second two are ff_USER_mask (say unable all ffs) and ps_mask, at the first boot. (see [here](https://github.com/apollo-lhc/cm_mcu/pull/145/files#diff-9cb461e4d0d68a2faa7e67efb69c308c2d463228064bb42ac14abd5e01af4a5fR862))

After that, it won't be read from the EEPROM again. Also, I think it's fine right? because the setFFmask() where the overwrite seems to take place is called only once implicitly from InitTask.c

Reply:
sorry, I think I see the issue now why you would like the ff_USER_mask always syncs with ff_mask in the internal EEPROM. For instance, the `set_id`  can change EEPROM while the USER mask is currently not updated and so FF won't be enabled accordingly. 
I will make sure that the two are in synced.

Reply:
ok, please let me know when this issue is resolved. also I now realize that you are storing the mask in a different way than the `ff_mask` in the internal EEPROM is stored; this is stored as a straight 20 bits w/o any gaps. I would like to unify to that mask convention.

Reply:
The latest commit resolve the comment about unconventional way of obtaining the ff PRESENT mask. 

For the synchronization of ff USER mask and the internal EEPROM, I have it `[here](https://github.com/apollo-lhc/cm_mcu/blob/first_time_mcu/projects/cm_mcu/commands/BoardCommands.c#L58-L60)` . Alternatively, we can add a line monitor task or somewhere to constantly reading the internal EEPROM and update the ff USER mask, or you may have other ideas?

Reply:
That works if someone sets a new value in the EEPROM but what happens if I set it the previous time it booted? In that case do you read the current value out of the EEPROM?

Reply:
no, whenever it's booted the ff USER mask will be forced to be the same as the ff PRESENT mask. I can change [this](https://github.com/apollo-lhc/cm_mcu/blob/83bd2722e6eb7843aad9878a3b6464b0313fd6f1/projects/cm_mcu/LocalTasks.c#L355) to read EEPROM ff mask directly then?